helper/resource/testing_new.go

```diff
@@ -442,3 +453,25 @@ func testIDRefresh(ctx context.Context, t testing.T, c TestCase, wd *plugintest.
 
 	return nil
 }
+
+func copyWorkingDir(ctx context.Context, t testing.T, stepNumber int, wd *plugintest.WorkingDir) {
+	if os.Getenv(plugintest.EnvTfAccPersistWorkingDir) == "" {
+		return
+	}
+
+	workingDir := wd.GetHelper().WorkingDirectory()
+	parentDir := filepath.Dir(workingDir)
+
+	dest := parentDir + "_" + strconv.Itoa(stepNumber)
+
+	err := plugintest.CopyDir(wd.GetHelper().WorkingDirectory(), dest)
+	if err != nil {
+		logging.HelperResourceError(ctx,
+			"Unexpected error copying working directory files",
+			map[string]interface{}{logging.KeyError: err},
+		)
+		t.Fatalf("TestStep %d/%d error copying working directory files: %s", stepNumber, err)
+	}
+
+	t.Log(fmt.Sprintf("Working directory and files have been copied to: %s", dest))
```

Comment:
Nit: https://pkg.go.dev/testing#T.Logf (surprised that this isn't caught by `go vet` -- maybe we should double check if there's an upstream issue for that?)

```suggestion
	t.Logf("Working directory and files have been copied to: %s", dest)
```